File: chat-backend/src/main/java/com/example/chat/message/dto/ChatMessageResponse.java
package com.example.chat.message.dto;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class ChatMessageResponse {

    private Long id;

    private Long userId;

    private String username;

    private String fullName;

    private Long recipientId;

    private LocalDateTime sentAt;

    private String content;

    private ReplyPreview replyTo;

}

File: chat-frontend/src/app/features/chat/models/chat-message-response.ts
import { ReplyPreview } from './reply-preview';

export interface ReactionSummary {
  emoji: string;
  count: number;
  reactedByMe: boolean;
  userIds: number[];
}

export interface ChatMessageResponse {
  id: number;

  userId: number;

  username: string;

  fullName: string;

  recipientId: number;

  content: string;

  sentAt: string;

  status: string;

  replyTo: ReplyPreview | null;

  edited?: boolean;
  editedAt?: string;
  deleted?: boolean;

  reactions?: ReactionSummary[];
}